File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/python3_solutions/1093-statistics-from-a-large-sample.py
from typing import List

class Solution:
    def sampleStats(self, count: List[int]) -> List[float]:
        min_val = -1
        max_val = -1
        total_sum = 0
        total_count = 0
        max_freq = 0
        mode_val = 0

        for x in range(256):
            c = count[x]
            if c > 0:
                if min_val == -1:
                    min_val = x
                max_val = x

                total_sum += x * c
                total_count += c

                if c > max_freq:
                    max_freq = c
                    mode_val = x

        if total_count == 0:
            return [0.0, 0.0, 0.0, 0.0, 0.0]

        mean_val = float(total_sum) / total_count

        median_val = 0.0
        current_count_sum = 0

        if total_count % 2 == 1:
            target_index = total_count // 2
            for x in range(256):
                current_count_sum += count[x]
                if current_count_sum > target_index:
                    median_val = float(x)
                    break
        else:
            target_index_1 = total_count // 2 - 1
            target_index_2 = total_count // 2

            val1 = -1
            val2 = -1

            for x in range(256):
                current_count_sum += count[x]
                if val1 == -1 and current_count_sum > target_index_1:
                    val1 = x
                if val2 == -1 and current_count_sum > target_index_2:
                    val2 = x
                    break
            median_val = (float(val1) + float(val2)) / 2.0

        return [float(min_val), float(max_val), mean_val, median_val, float(mode_val)]

File: leetcode/AIcoding/1.Gemini-2.5/rust_solutions/1093-statistics-from-a-large-sample.rs
impl Solution {
    pub fn sample_statistics(count: Vec<i32>) -> Vec<f64> {
        let mut min_val: f64 = 0.0;
        let mut max_val: f64 = 0.0;
        let mut mean_val: f64 = 0.0;
        let mut median_val: f64 = 0.0;
        let mut mode_val: f64 = 0.0;

        let mut total_count: i64 = 0;
        let mut sum_values: i64 = 0;
        let mut max_freq: i32 = 0;

        let mut first_val_found = false;
        for i in 0..=255 {
            let freq = count[i as usize];
            if freq > 0 {
                if !first_val_found {
                    min_val = i as f64;
                    first_val_found = true;
                }
                max_val = i as f64;

                total_count += freq as i64;
                sum_values += (i as i64) * (freq as i64);

                if freq > max_freq {
                    max_freq = freq;
                    mode_val = i as f64;
                }
            }
        }

        if total_count == 0 {
            return vec![0.0, 0.0, 0.0, 0.0, 0.0];
        }

        mean_val = sum_values as f64 / total_count as f64;

        let mut current_rank: i64 = 0;
        let mut median_val1: f64 = 0.0;
        let mut median_val2: f64 = 0.0;
        let mut found_median1 = false;
        let mut found_median2 = false;

        let target_rank1 = (total_count + 1) / 2;
        let target_rank2 = (total_count + 2) / 2;

        for i in 0..=255 {
            let freq = count[i as usize];
            if freq > 0 {
                if !found_median1 && (current_rank + freq as i64) >= target_rank1 {
                    median_val1 = i as f64;
                    found_median1 = true;
                }
                if !found_median2 && (current_rank + freq as i64) >= target_rank2 {
                    median_val2 = i as f64;
                    found_median2 = true;
                }

                current_rank += freq as i64;

                if found_median1 && found_median2 {
                    break;
                }
            }
        }

        if total_count % 2 == 1 {
            median_val = median_val1;
        } else {
            median_val = (median_val1 + median_val2) / 2.0;
        }

        vec![min_val, max_val, mean_val, median_val, mode_val]
    }
}